### ISSUE TITLE
It is a dangerous trap to mix and match `insert` and `insert_str`

### ISSUE THREAD

**Issue by reivilibre** [sile/patricia_tree#27]
The `insert_str` methods in #15 seem to lead to a trap/footgun: if you mix and match `insert` and `insert_str`, then depending on what exactly you did, sometimes you won't be able to `get` things again afterwards.

Example:

```rust
use patricia_tree::PatriciaMap;

pub fn main() {
    let mut pmap = PatriciaMap::new();
    pmap.insert("ab/»", true);
    pmap.insert("ab/¡", true);
    pmap.insert_str("ab/«", true);

    assert!(*pmap.get("ab/«").unwrap()); // <--- this unwrap fails
    assert!(*pmap.get("ab/»").unwrap());
    assert!(*pmap.get("ab/¡").unwrap());
}
```

The motivation for `insert_str` makes sense and I think I can imagine, based on the tree structure, why this issue occurs. Maybe it would have been better for a stringy-PatriciaMap to be a different type, that ensures you can't mix them up?

I don't know whether it is interesting to you or not to break backwards compat for the sake of distinguishing these at the type system level, but it felt like it was worth mentioning. I only realised this issue after I introduced so many `insert`s that I couldn't actually track down where I was inserting wrong, so in the end I replaced all my `insert_str` with `insert` since they're easier to search for in a code base that also uses other kinds of maps :).

**Comment by sile** [comment]
Thank you for reporting this issue. I agree that this is an important problem that should be fixed.
Introducing another type for  stringy-PatriciaMap sounds good idea. 
This weekend I'll take a closer look at how to fix this problem.

**Comment by sile** [comment]
Addressed this issue and released v0.6.0. Thanks again!
